src/west/manifest.py

```diff
@@ -1422,6 +1483,85 @@ def _load(self, manifest: Dict[str, Any],
 
         _logger.debug(f'loaded {loading_what}')
 
+    def _load_groups(self, manifest: Dict[str, Any]):
+        # Reset self.groups and self.no_groups if the 'manifest' data
+        # or the local configuration file (just local, not global or
+        # system) have nonempty values for them.
+
+        cfg_groups, cfg_no_groups = \
+            self._load_config_groups()
+
+        manifest_groups, manifest_no_groups = \
+            self._load_manifest_groups(manifest)
+
+        if cfg_groups or manifest_groups:
+            cfg_groups = cfg_groups or []
+            manifest_groups = manifest_groups or []
+            self.groups = cfg_groups + manifest_groups
+
+        if cfg_no_groups or manifest_no_groups:
+            cfg_no_groups = cfg_no_groups or []
+            manifest_no_groups = manifest_no_groups or []
+            self.no_groups = cfg_no_groups + manifest_no_groups
```

Comment:
So if manifest contains:
```
    groups:
        - groupA
        - groupB
      no-groups:
        - groupC
```
and user configures:
```
[manifest]
#  I want group C
    groups = groupC
#  I don't want group A and B
    no-groups = groupA,groupB
```
then the `self.groups` and `self.no_groups` will both contain:
```
['groupA', 'groupB', 'groupC']
```

I don't think this is the intention.

When fixing this, I think user defined group configs should have precedence over manifest defaults.

Reply:
> I don't think this is the intention.

It is the intention; what's the issue?

Reply:
I have made a small manifest example with groups here:
https://github.com/tejlmand/zephyr/tree/testing/west_454_groups

The manifest projects are divided into three groups: `groupA`, `groupB`, `groupC`.

`groupA` and `groupB` has a manifest setup, and `groupC` is just default, as:
https://github.com/tejlmand/zephyr/blob/7f42bb41c77d4d6d48930b211ca65ca8da983a65/west.yml#L17-L21
```
manifest:
  groups:
    - groupA
  no-groups:
    - groupB
```

doing a `west list` I get:
```
$ west list
manifest     zephyr                       HEAD                                     N/A
cmsis        modules/hal/cmsis            542b2296e6d515b265e25c6b7208e8fea3014f90 https://github.com/zephyrproject-rtos/cmsis
```
just as expected.

Now, I don't want `groupA` listed (or updated), so let me disable that:
```
$ west config manifest.no-groups groupA
$ west list
manifest     zephyr                       HEAD                                     N/A
cmsis        modules/hal/cmsis            542b2296e6d515b265e25c6b7208e8fea3014f90 https://github.com/zephyrproject-rtos/cmsis
```
Hooh, it's still there. As a user I want to be able to overrule the manifest file and disable a group.

Let me see if I can enable `groupB`.
```
$ west config manifest.groups groupB
$ west list
manifest     zephyr                       HEAD                                     N/A
cmsis        modules/hal/cmsis            542b2296e6d515b265e25c6b7208e8fea3014f90 https://github.com/zephyrproject-rtos/cmsis
hal_atmel    modules/hal/atmel            9f0a699a2f0253dc9688211ef696daa7b8fadb75 https://github.com/zephyrproject-rtos/hal_atmel
```
working as expected.

(Same happens for `west update` of course)

Also:
```
$ west help update
...
advanced options:
  ....
  -G GROUPS, --no-groups GROUPS
                        update as if GROUPS are in manifest.no-groups
```
Cool, let me try this:
```
$ west update -G groupA
=== updating cmsis (modules/hal/cmsis):
HEAD is now at 542b229 DSP: Integrate CMSIS-DSP 1.8.0 (CMSIS 5.7.0)
```
Damn, `groupA` is still being updated.

Note, it's not only the commented line with this, but both those lines:
https://github.com/mbolivar-nordic/west/blob/project-groups/src/west/manifest.py#L1506
https://github.com/mbolivar-nordic/west/blob/project-groups/src/west/manifest.py#L1511

I believe the user should be able to overrule settings from `west.yml` or `.west/config`

Reply:
>Hooh, it's still there. As a user I want to be able to overrule the manifest file and disable a group.

+1 to that.

>Damn, groupA is still being updated.

That's a bug IMHO. `-G` should take precedence to configs and manifest, since the user is overriding explicitly.